uiowa/uiowa
# Events block date range

Resolves https://github.com/uiowa/uiowa/issues/5187
Resolves https://github.com/uiowa/uiowa/issues/4320

# Test

- `ddev composer install && ddev blt frontend && ddev blt ds --site=sandbox.uiowa.edu`
- Inspect https://sandbox.uiowa.ddev.site/node/686 to make sure they are showing upcoming and past blocks accurately and that min and max attributes on the input elements are set (which limits the date picker).
- Sync admissions and grad to make sure event and thesis defense content types import correctly and the smart date functionality still works.
- The past date checkbox on the v3 events block should be hidden and there should now be three fields to replace it (timeframe, sort, date range (hidden until date range is selected from timeframe).
- Check that the date range field is appropriately displayed with no overlap of the elements.
- Field validation should occur on the date range where it requires values if timeframe:date range is selected and that the values themselves do not extend the min and max.
- Help text should make things relatively clear though I understand Asc/Desc is complicated in a upcoming/past/mix context.
- Inspect code to confirm that it is limited to v3 functionality and does not affect v2 paragraphs logic.

# Todo

- [x] Validation - require and within safe limits
- [x] Limit date picker to min and max. Seems to require min/max attributes being present on the widget.
- ~~Adjust the logic associated with the past date update hook. - It works in situations where the field is set and has a value, but we thought we should check for instances where the field is not set (neither true/false). If not set, it should also be set to upcoming. https://github.com/uiowa/uiowa/pull/5497/files#diff-fead566a578ea17efbda9b82e2a399962f8d578c39998fedf5bb87ceebd7d3f6R2139~~ _since we are checking for 1, and covering 0 and null with the else, no separate check is necessary._
- [x] Set date range field during update hook or set min/max attributes if not present.
- [x] Form styles for date range field - right now they don't display with enough spacing between elements. @makurj 
- [x] Help text for the new fields (sort, timeframe and date range)
- [x] Clean up smart date config from feature and site splits - check for sites that have smart date split, remove the declaration from the split, remove the smart_date configuration unless it is different than what is in default. if different, a partial export is needed.
- [x] Remove out of date `@todos`, convert any to separate issues that can't be removed.
- [x] If https://github.com/uiowa/uiowa/pull/5501 is merged first, remove smart date from iowa now site split.

# Follow-up
- https://github.com/uiowa/uiowa/issues/5521
- https://github.com/uiowa/uiowa/issues/5518
